config/application.rb

```diff
@@ -22,5 +22,9 @@ class Application < Rails::Application
 
     config.exceptions_app = routes
     config.autoload_paths += %W[#{config.root}/app/loggers lib/traject]
+
+    # refer to https://github.com/projectblacklight/blacklight/issues/2768
+    # refer to https://github.com/projectblacklight/blacklight/pull/2770/files
+    config.active_record.yaml_column_permitted_classes = [ActiveSupport::HashWithIndifferentAccess, Symbol]
```

Comment:
@bwatson78 This is a backward compatibility fix that ensures the upgrade to Rails 5.2 does not break search results. Please refer to the resources in the comments for more details. I relied on the PR for v7.28.0 of Blacklight to add the fix above.